### PR TITLE
Dev/red

### PR DESCRIPTION
**Pull request by ens-carlos**
Added Red (REpeat Detector) analysis to the genome annotation config file including updates to the create annotation config script and parameters to replace repeatmask repbase with repeatdetector features for softmasking the genome.

### CODE REVIEW

File: modules/Bio/EnsEMBL/Analysis/Hive/Config/Genome_annotation_conf.pm [hunk 1]
```diff
@@ -134,6 +135,9 @@ sub default_options {
 # Pipe and ref db info
 ########################
 
+    'red_logic_name'            => 'repeatdetector', # logic name for the Red repeat finding analysis
+
+    'projection_source_db_name'    => 'homo_sapiens_core_100_38', # This is generally a pre-existing db, like the current human/mouse core for example
```
Comment (thibauthourlier) [inline modules/Bio/EnsEMBL/Analysis/Hive/Config/Genome_annotation_conf.pm:140]: This should not be in the PR

File: modules/Bio/EnsEMBL/Analysis/Hive/Config/Genome_annotation_conf.pm [hunk 2]
```diff
@@ -1572,7 +1590,7 @@ sub pipeline_analyses {
               '(1, "annotation.provider_name", "'.$self->o('annotation_provider_name').'"),'.
               '(1, "annotation.provider_url", "'.$self->o('annotation_provider_url').'"),'.
               '(1, "species.production_name", "'.$self->o('production_name').'"),'.
-              '(1, "repeat.analysis", "'.$self->o('full_repbase_logic_name').'"),'.
+	      (!($self->o('replace_repbase_with_red_to_mask')) ? '(1, "repeat.analysis", "'.$self->o('full_repbase_logic_name').'"),': '').
```
Comment (thibauthourlier) [inline modules/Bio/EnsEMBL/Analysis/Hive/Config/Genome_annotation_conf.pm:1593]: RED will still need to have this meta key inserted

**Review by thibauthourlier — COMMENTED** [review]
I think all the hard work should happen in the config script, not in the config itself. It might mean a change of parameter name to be more generic or adding a generic parameter which is then set by `full_repbase_logic_name` or `red_logic_name`

I am not formatlly requesting a change because I cannot review it in the next weeks...

File: modules/Bio/EnsEMBL/Analysis/Hive/Config/Genome_annotation_conf.pm (hunk 1 above)
Comment (ens-carlos) [reply modules/Bio/EnsEMBL/Analysis/Hive/Config/Genome_annotation_conf.pm]: I've just removed the projection_source_db_name initialisation as it's also a duplicate.

File: modules/Bio/EnsEMBL/Analysis/Hive/Config/Genome_annotation_conf.pm (hunk 2 above)
Comment (ens-carlos) [reply modules/Bio/EnsEMBL/Analysis/Hive/Config/Genome_annotation_conf.pm]: Just added it.

**Review by thibauthourlier — APPROVED** [review]
Something annoys me in the way it is set up but I don't see an easy way to do what I would like

**Pull request merged**